### ISSUE TITLE
Add ability to handle win32 relative paths

### ISSUE THREAD

**Issue by davakh** [webpack/enhanced-resolve#401]
### Issue
Right now, based on codebase, library supports only posix relative paths. 

### Reproduction
This is **correct** relative path in context of this library:
```
../../../src
```

This is **incorrect** relative path in context of this library:
```
..\\..\\..\\src
```

To understand how the library understands these paths, you can call `getType` from [this file](https://github.com/webpack/enhanced-resolve/blob/58464fc/lib/util/path.js#L40).

So, anything that can be passed as a result by [path.relative(...)](https://nodejs.org/api/path.html#pathrelativefrom-to) in Node.js, will not work as expected on Windows machines. 

### Solutions

Possible solution is to add separation in the enum of types for `RelativePosix` and `RelativeWin`, like for the `Absolute..` paths. But it's hard to predict what integrations with this library are going to break, so I think it can be a big deal. 

### Workaround

Add replacing for backward compatibility with Windows (Windows can handle posix separator):
```
path.relative(....).split(path.win32.sep).join(path.posix.sep);
```

----------

**PS.**  Maybe it's the only way you want to handle paths in this library, and then I think it would be cool to add information about this scenario or to close this issue with a preferable workaround.

**Comment by alexander-akait** [comment]
Hello, can you provide a problem/example where you got such paths?

**Comment by davakh** [comment]
It's possible to get this type of paths when you work with [module-federation](https://github.com/module-federation/universe) f.e. Of course, it's overhead to give an example with another library, but I can provide an example with current library via [codesandbox](https://codesandbox.io/p/devbox/example-of-relative-paths-in-webpack-enhanced-resolve-yvnw56?layout=%257B%2522sidebarPanel%2522%253A%2522EXPLORER%2522%252C%2522rootPanelGroup%2522%253A%257B%2522direction%2522%253A%2522horizontal%2522%252C%2522contentType%2522%253A%2522UNKNOWN%2522%252C%2522type%2522%253A%2522PANEL_GROUP%2522%252C%2522id%2522%253A%2522ROOT_LAYOUT%2522%252C%2522panels%2522%253A%255B%257B%2522type%2522%253A%2522PANEL_GROUP%2522%252C%2522contentType%2522%253A%2522UNKNOWN%2522%252C%2522direction%2522%253A%2522vertical%2522%252C%2522id%2522%253A%2522clqzvk4ro0006356kwhphk3pe%2522%252C%2522sizes%2522%253A%255B70%252C30%255D%252C%2522panels%2522%253A%255B%257B%2522type%2522%253A%2522PANEL_GROUP%2522%252C%2522contentType%2522%253A%2522EDITOR%2522%252C%2522direction%2522%253A%2522horizontal%2522%252C%2522id%2522%253A%2522EDITOR%2522%252C%2522panels%2522%253A%255B%257B%2522type%2522%253A%2522PANEL%2522%252C%2522contentType%2522%253A%2522EDITOR%2522%252C%2522id%2522%253A%2522clqzvk4ro0002356k14viyzjs%2522%257D%255D%257D%252C%257B%2522type%2522%253A%2522PANEL_GROUP%2522%252C%2522contentType%2522%253A%2522SHELLS%2522%252C%2522direction%2522%253A%2522horizontal%2522%252C%2522id%2522%253A%2522SHELLS%2522%252C%2522panels%2522%253A%255B%257B%2522type%2522%253A%2522PANEL%2522%252C%2522contentType%2522%253A%2522SHELLS%2522%252C%2522id%2522%253A%2522clqzvk4ro0004356kpg7xibxe%2522%257D%255D%252C%2522sizes%2522%253A%255B100%255D%257D%255D%257D%252C%257B%2522type%2522%253A%2522PANEL_GROUP%2522%252C%2522contentType%2522%253A%2522DEVTOOLS%2522%252C%2522direction%2522%253A%2522vertical%2522%252C%2522id%2522%253A%2522DEVTOOLS%2522%252C%2522panels%2522%253A%255B%257B%2522type%2522%253A%2522PANEL%2522%252C%2522contentType%2522%253A%2522DEVTOOLS%2522%252C%2522id%2522%253A%2522clqzvk4ro0005356knd7jgmc4%2522%257D%255D%252C%2522sizes%2522%253A%255B100%255D%257D%255D%252C%2522sizes%2522%253A%255B100%252C0%255D%257D%252C%2522tabbedPanels%2522%253A%257B%2522clqzvk4ro0002356k14viyzjs%2522%253A%257B%2522tabs%2522%253A%255B%257B%2522id%2522%253A%2522clqzvk4ro0001356kbvve1pnh%2522%252C%2522mode%2522%253A%2522permanent%2522%252C%2522type%2522%253A%2522FILE%2522%252C%2522filepath%2522%253A%2522%252FREADME.md%2522%252C%2522state%2522%253A%2522IDLE%2522%257D%255D%252C%2522id%2522%253A%2522clqzvk4ro0002356k14viyzjs%2522%252C%2522activeTabId%2522%253A%2522clqzvk4ro0001356kbvve1pnh%2522%257D%252C%2522clqzvk4ro0005356knd7jgmc4%2522%253A%257B%2522tabs%2522%253A%255B%255D%252C%2522id%2522%253A%2522clqzvk4ro0005356knd7jgmc4%2522%257D%252C%2522clqzvk4ro0004356kpg7xibxe%2522%253A%257B%2522id%2522%253A%2522clqzvk4ro0004356kpg7xibxe%2522%252C%2522tabs%2522%253A%255B%257B%2522id%2522%253A%2522clqzvk4ro0003356khjbkclsg%2522%252C%2522mode%2522%253A%2522permanent%2522%252C%2522type%2522%253A%2522TASK_LOG%2522%252C%2522taskId%2522%253A%2522start%2522%257D%255D%252C%2522activeTabId%2522%253A%2522clqzvk4ro0003356khjbkclsg%2522%257D%257D%252C%2522showDevtools%2522%253Afalse%252C%2522showShells%2522%253Atrue%252C%2522showSidebar%2522%253Atrue%252C%2522sidebarPanelSize%2522%253A34.4921875%257D).

**Comment by alexander-akait** [comment]
@davakh Sorry for delay, do you want to send a PR?

**Comment by davakh** [comment]
Yes, I have some thoughts on how to solve it. I'll try to give a PR in near future.

**Comment by xiaoxiaojx** [comment]
Sorry for missing this message! This is a subtle issue — passing Node.js path.relative() directly to the resolver works on POSIX but fails on Windows. If the change is small and doesn’t introduce any performance regressions, it might be worth supporting, since the path comes from the official path.relative() API. However, it seems that both oxc-resolver and rspack-resolver don’t support this either. https://github.com/webpack/enhanced-resolve/actions/runs/25276455827/job/74107022461?pr=564.
